### PR TITLE
tests: static-server can continue if already running

### LINKED ISSUE #14302: `yarn smoke` should work when `yarn static-server` is first called manually
https://github.com/GoogleChrome/lighthouse/pull/14289#discussion_r946204984

### PR DESCRIPTION
**Pull request by paulirish**
fixes #14302

not trying to get fancy. these are pretty unique ports.

### CODE REVIEW

File: cli/test/fixtures/static-server.js [hunk 1]
```diff
@@ -249,20 +250,26 @@ class Server {
...
   });
-  await Promise.all(servers.map(s => s.listen(s._port, 'localhost')));
+
+  await Promise.allSettled(servers.map(s => s.listen(s._port, 'localhost'))).then(outcomes => {
+    if (outcomes.every(o => o.status === 'fulfilled')) return;
+    if (outcomes.every(o => o.reason.message.includes('already'))) {
+      return console.warn('😧 Server already up. Continuing…');
+    }
+    console.error(outcomes.map(o => o.reason));
+  });
```
Comment (adamraine) [inline cli/test/fixtures/static-server.js:266]: For readability, will this work?

Also we should probably throw an error if the servers fail for an unknown reason

```suggestion
  const outcomes = await Promise.allSettled(servers.map(s => s.listen(s._port, 'localhost')));

  if (outcomes.some(o => o.status === 'rejected')) {
    if (outcomes.every(o => o.reason.message.includes('already'))) {
      console.warn('😧 Server already up. Continuing…');
    } else {
      console.error(outcomes.map(o => o.reason));
      throw new Error('One or more servers did not start correctly');
    }
  }

  return servers
```

**Review by connorjclark — APPROVED** [review]
Approval w/ Adam's suggestions.

File: cli/test/fixtures/static-server.js (hunk 1 above)
Comment (paulirish) [reply cli/test/fixtures/static-server.js:266]: your await flow is definitely better. and fixes the error throwing. (i was ending up with an unhandled rejection previously)

i do still prefer the style of my two every()s, but whatever. definitely not a big deal.

Comment (paulirish) [reply cli/test/fixtures/static-server.js:266]: @adamraine thanks for writing out the suggestion! :) so clean.

**Pull request merged**